client/my-sites/email/email-providers-comparison/price-with-interval/index.tsx

```diff
@@ -1,11 +1,30 @@
 import formatCurrency from '@automattic/format-currency';
 import { translate } from 'i18n-calypso';
+import { hasIntroductoryOfferFreeTrial } from 'calypso/lib/emails';
 import { IntervalLength } from 'calypso/my-sites/email/email-providers-comparison/interval-length';
 import type { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
 import type { ReactElement } from 'react';
 
 import './style.scss';
 
+const getSalePrice = ( {
+	isEligibleForIntroductoryOffer,
+	isDiscounted,
+	product,
+}: {
+	isEligibleForIntroductoryOffer: boolean;
+	isDiscounted: boolean;
+	product: ProductListItem | null;
+} ) => {
+	if ( hasIntroductoryOfferFreeTrial( product ) ) {
+		return 0;
+	}
+	if ( isEligibleForIntroductoryOffer && ! isDiscounted ) {
+		return product?.introductory_offer?.cost_per_interval ?? 0;
+	}
```

Comment:
> I'm also wondering if we can lose the ?? checks

I'd leave those in there because the `product` object may be undefined in between render phases. The tests for discount and the retrieval of the `sale_cost` are unrelated technically as they refer to different properties of the `product`.  Also TS syntactically requires it.

>  I'm also wondering if we should return null instead of 0 

This is moot, as the function has been simplified.